thrift/thrift0.13/src/test/java/com/linecorp/armeria/server/thrift/THttpServiceBlockingTest.java

```diff
@@ -0,0 +1,108 @@
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+import testing.thrift.test.TestService;
+import testing.thrift.test.TestServiceRequest;
+import testing.thrift.test.TestServiceResponse;
+
+class THttpServiceBlockingTest {
+    private static final AtomicBoolean blocking = new AtomicBoolean();
+
+    private static final ScheduledExecutorService executor =
+            new ScheduledThreadPoolExecutor(1, ThreadFactories.newThreadFactory("blocking-test", true)) {
+                @Override
+                protected void beforeExecute(Thread t, Runnable r) {
+                    blocking.set(true);
+                }
+            };
+
+    @BeforeEach
+    void clear() {
+        blocking.set(false);
+    }
+
+    @RegisterExtension
+    static final ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+
+            sb.service("/",
+                       THttpService.builder().addService(new TestServiceImpl()).build());
+            sb.service("/blocking", THttpService.builder()
+                                                .useBlockingTaskExecutor(true)
+                                                .addService(new TestServiceImpl())
+                                                .build());
+            sb.blockingTaskExecutor(executor, true);
+        }
+    };
+
+    @Test
+    void nonBlocking() throws Exception {
+        final TestService.Iface client = ThriftClients.newClient(server.httpUri(), TestService.Iface.class);
+
+        final String message = "nonBlockingTest";
+        final TestServiceResponse response = client.get(new TestServiceRequest(message));
+
+        assertThat(response.response).isEqualTo(message);
+        assertThat(blocking).isFalse();
+    }
+
+    @Test
+    void blocking() throws Exception {
+        final TestService.Iface client =
+                ThriftClients.builder(server.httpUri())
+                             .decorator((delegate, ctx, req) -> {
+                                 final HttpRequest newReq = req.mapHeaders(
+                                         headers -> headers.toBuilder().path("/blocking").build());
+                                 ctx.updateRequest(newReq);
+                                 return delegate.execute(ctx, newReq);
+                             })
+                             .build(TestService.Iface.class);
+        final String message = "blockingTest";
+        final TestServiceResponse response = client.get(new TestServiceRequest(message));
+
+        assertThat(response.response).isEqualTo(message);
+        assertThat(blocking).isTrue();
+    }
+
+    static class TestServiceImpl implements TestService.AsyncIface {
+        @Override
+        public void get(TestServiceRequest request,
+                        AsyncMethodCallback resultHandler) throws TException {
+            resultHandler.onComplete(new TestServiceResponse(request.getMessage()));
```

Comment:
- Could we check whether the current thread is the blocking task executor?
- Should we also check `TestService.Iface`?

Reply:
> Could we check whether the current thread is the blocking task executor?
I checked it here
https://github.com/line/armeria/pull/5619/files#diff-e2ea70c1bdf1471b98c09d0172aabae667532ea1b8f36b99bfd9c1011d38ed34R99

> Should we also check TestService.Iface?
Fixed it. Because of Iface, only verified the current thread is blocking executor without adding `useBlockingTaskExecutor `